gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/ImmunizationMapper.java

```diff
@@ -0,0 +1,301 @@
+                return DateFormatUtil.parseToDateTimeType(low).getValue();
+            } else if (low != null) {
+                return DateFormatUtil.parseToDateTimeType(low).getValue();
+            } else if (high != null && availabilityTimeValue.getValue() != null && availabilityTimeValue.getNullFlavor() == null) {
+                return DateFormatUtil.parseToDateTimeType(availabilityTimeValue.getValue()).getValue();
+            }
+        }
+
+        return null;
+    }
+
+    private String highValueToNotes(RCMRMT030101UK04ObservationStatement observationStatement) {
+        if (observationStatement.getEffectiveTime() != null) {
+            var low = getIVXBTSStringValue(observationStatement.getEffectiveTime().getLow());
+            var high = getIVXBTSStringValue(observationStatement.getEffectiveTime().getHigh());
+            var availabilityTimeValue = observationStatement.getAvailabilityTime();
+
+            if (low != null && high != null) {
+                return END_DATE_PREFIX + high;
+            } else if (high != null && availabilityTimeValue.getValue() != null) {
+                return END_DATE_PREFIX + high;
+            } else if (high != null && availabilityTimeValue.getNullFlavor() != null) {
+                return END_DATE_PREFIX + high;
+            }
+        }
+
+        return null;
+    }
+
+    private Annotation buildNote(RCMRMT030101UK04ObservationStatement observationStatement) {
+        var pertinentText = getPertinentAnnotation(observationStatement);
+        var dateText = highValueToNotes(observationStatement);
+
+        if (pertinentText != null && dateText != null) {
+            var noteStringType = new StringType(pertinentText + StringUtils.SPACE + dateText);
+            return new Annotation(noteStringType);
+        } else if (pertinentText != null) {
+            var noteStringType = new StringType(pertinentText);
+            return new Annotation(noteStringType);
+        } else if (dateText != null) {
+            var noteStringType = new StringType(dateText);
+            return new Annotation(noteStringType);
+        }
+
+        return null;
+    }
+
+    private String getPertinentAnnotation(RCMRMT030101UK04ObservationStatement observationStatement) {
+        if (observationStatement.getPertinentInformation() != null && observationStatement.getPertinentInformation().size() > 0) {
+            var pertinentInformation = observationStatement.getPertinentInformation().get(0);
+
+            if (pertinentInformation.getPertinentAnnotation() != null && pertinentInformation.getPertinentAnnotation().getText() != null) {
+                return pertinentInformation.getPertinentAnnotation().getText();
+            }
+        }
+
+        return null;
+    }
+
+    private Immunization createImmunization(ImmunizationMapperParameters immunizationMapperParameters) {
+        var immunization = new Immunization();
+
+        immunization.getMeta().getProfile().add(new UriType(META_PROFILE));
```

Comment:
Since we're working on a new Immunization object and won't have any data in the meta field, we can just set the profile here instead of getting & then adding to it.